src/gui/toolbarMenubar/model/ColorPalette.cpp

```diff
@@ -0,0 +1,154 @@
+#include "ColorPalette.h"
+
+#include <fstream>
+#include <limits>
+#include <sstream>
+
+#include <gtk/gtk.h>
+#include <util/StringUtils.h>
+
+Palette::Palette(fs::path path): filepath{std::move(path)}, header{}, namedColors{} {}
+
+void Palette::load() {
+    header.clear();
+    namedColors.clear();
+
+    std::ifstream gptFile{filepath};
+    std::string line;
+
+    if (gptFile.is_open()) {
+        getline(gptFile, line);
+        if (!parseFirstGimpPaletteLine(line))
+            return;
+        while (getline(gptFile, line)) {
+            parseHeaderLine(line) || parseColorLine(line);
+        }
+    }
+}
+
+auto Palette::parseFirstGimpPaletteLine(const std::string& line) -> bool {
+    if (StringUtils::trim(line) != "GIMP Palette") {
+        g_error(".gpt file needs to start with \"GIMP Palette\" in the "
+                "first line");
+        return false;
+    }
+    return true;
+}
+
+auto Palette::parseHeaderLine(const std::string& line) -> bool {
+    Header headerEntry;
+    NamedColor color;
+    std::istringstream iss{line};
+    if (iss >> headerEntry) {
+        header[headerEntry.attribute] = headerEntry.value;
+        return true;
+    }
+    return false;
+}
+
+auto Palette::parseColorLine(const std::string& line) -> bool {
+    NamedColor color;
+    std::istringstream iss{line};
+    if (iss >> color) {
+        // call to private variable paletteIndex allowed because of friend class
+        color.paletteIndex = namedColors.size();
+        namedColors.push_back(color);
+        return true;
+    }
+    return false;
+}
+
+
+NamedColor Palette::getColorAt(size_t i) const {
```

Comment:
```suggestion
NamedColor const& Palette::getColorAt(size_t i) const {
```

Reply:
You meant `const NamedColor& Palette::getColorAt(size_t i) const`, right?
That is definitely way prettier. Thanks :pray: !

Reply:
There is no difference between `T const&` and `const T&`. It's completely a matter of flavor. I prefer the first, because a preceding `const` only exists before "named types". Thus, the first version is in my opinion more consistent/consequent. E.g.:
`T const* const* const` `==` `const T* const* const`

Reply:
Thanks for explaining. I looked it up and it seems that there are some other benefits to this way of writing it ([stackoverflow](https://stackoverflow.com/questions/2640446/why-do-some-people-prefer-t-const-over-const-t)). I'll try it out.